aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -40,53 +76,116 @@ std::tuple<at::Tensor,at::Tensor,at::Tensor> mkldnn_convolution_backward(
 
 using namespace mkldnn;
 
-namespace {
-// Helper function for getting an ideep tensor out of an aten Tensor.
-// Note in case the aten Tensor is a dense tensor, the retured ideep
-// tensor is just a view of the storage of the aten dense tensor, so
-// caller needs to make sure the aten dense tensor's lifetime is
-// longer than the ideep tensor.
-inline ideep::tensor get_mkldnn_tensor(const at::Tensor& tensor) {
-  if (tensor.is_mkldnn()) {
-    return at::native::itensor_from_mkldnn(tensor);
+namespace at { namespace native {
+
+using attr_t = ideep::descriptor_group::attr_t;
+using alloc = AllocForMKLDNN;
+
+static Tensor dnnl_conv2d_base(
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    IntArrayRef stride,
+    IntArrayRef padding,
+    IntArrayRef dilation,
+    int64_t groups,
+    attr_t attr_) {
+  // transfer Varibale problem to Tensor?
+  const ideep::tensor x = get_mkldnn_tensor(input);
+  const ideep::tensor w = get_mkldnn_tensor(weight);
+  c10::optional<ideep::tensor> b{c10::nullopt};
+  if (bias.numel() != 0) {
+    b = get_mkldnn_tensor(bias);
+  }
+
+  std::vector<int64_t> kernel_size(x.ndims());
+  if (w.ndims() == x.ndims() + 1) {
+    AT_ASSERTM(groups > 1,
+        "Only group dnnl_conv2d weights could have been reordered to 5d");
+    kernel_size[0] = w.get_dim(0) * w.get_dim(1);
+    std::copy_n(w.get_dims().cbegin() + 2, x.ndims() - 1,
+                kernel_size.begin() + 1);
   } else {
-    return at::native::itensor_view_from_dense(tensor);
+    std::copy_n(w.get_dims().cbegin(), x.ndims(), kernel_size.begin());
   }
-}
+
+  const ideep::param::dims x_dims = x.get_dims();
+  std::vector<int64_t> input_size{x_dims.cbegin(), x_dims.cend()};
+  std::vector<int64_t> output_sizes = conv_output_size(
+      input_size, kernel_size, padding, stride, dilation);
+
+  ideep::tensor y;
+  if (b.has_value()) {
+    ideep::convolution_forward::compute<alloc>(
+        x,
+        w,
+        b.value(),
+        {output_sizes.cbegin(), output_sizes.cend()},
+        y,
+        {stride.begin(), stride.end()},
+        {dilation.begin(), dilation.end()},
+        {padding.begin(), padding.end()},
+        {padding.begin(), padding.end()},
+        groups,
+        attr_,
+        ideep::algorithm::convolution_direct,
+        ideep::prop_kind::forward_inference);
```

Comment:
It is either "forward" or "forward_inference" depending on who calls it, e.g. conv_relu fusion only applicable in inference while conv better in "forward" training mode aligning current code behavior. Suggest to use a flag here.